src/Directory.Build.targets

```diff
@@ -34,18 +34,24 @@
           Condition="$(NeedsPublishing) == 'true'"
           DependsOnTargets="$(_BeforePublishNoBuildTargets);$(_CorePublishTargets)" />
 
-  <!-- Creates a file for packaged projects that declare a blog group prefix so that
-      publishing can use it in the blob path calculation. -->
-  <Target Name="GenerateBlobGroupFile"
+  <!-- Creates artifact files related to the package that will be uploaded to blob storage during publish. -->
+  <Target Name="GeneratePackageArtifactFiles"
           AfterTargets="Pack"
           Condition="'$(IsPackable)' == 'true' and '$(BlobGroupPrefix)' != ''">
     <PropertyGroup>
       <_BlobGroupVersionMajor>$(PackageVersion.Split('.')[0])</_BlobGroupVersionMajor>
       <_BlobGroupVersionMinor>$(PackageVersion.Split('.')[1])</_BlobGroupVersionMinor>
       <_BlobGroupName>$(BlobGroupPrefix)$(_BlobGroupVersionMajor).$(_BlobGroupVersionMinor)</_BlobGroupName>
     </PropertyGroup>
+    <!-- A file that contains the blob group so that publishing can use it in the blob path calculation. -->
     <WriteLinesToFile File="$(PackageOutputPath)\$(PackageId).$(PackageVersion).nupkg.blobgroup"
                       Lines="$(_BlobGroupName)"
                       Overwrite="true" />
+    <!-- A file that contains the package version so that aka.ms latest links can point to the latest
+         version within a blob group. -->
+    <WriteLinesToFile File="$(PackageOutputPath)\$(PackageId).$(PackageVersion).nupkg.version"
```

Comment:
Most repos follow the convention of using a SHA commit and a version for this, any reason not to follow suit?:

![image](https://user-images.githubusercontent.com/19413848/84579649-47cb1000-ad84-11ea-96d6-58348a8c6ddf.png)

Reply:
That's typically done to include the version/commit id file inside of whatever package is being shipped. The intent for what I'm doing here is to to produce a simple version file that will be uploaded to some blob storage (different that the storage the checksum is uploaded to) so that it can be used to look up (1) the latest version of the 5.0.X nuget package, and (2) to find the corresponding checksum file in the checksum blob feed (dotnetclichecksums).

Another reason to keep them separate is that the dotnet-docker folks typically recommend that these types of information should be stored in separate places in order to help mitigate malicious payloads. If the two pieces of information are together, someone malicious could theoretically upload the file that has a checksum and a version that corresponds with the malicious package. Then we'd look up whatever package that was. If they are in separate feeds, the malicious attacker would have to compromise all three parts (the checksum, the version, and then package). If any of those are out of alignment, then the automatic update would fail in the dotnet-docker repository and we'd get a clear indication something went wrong.

Reply:
@mthalman, @MichaelSimons what do think about having the checksum and version in the same file? Also, if I don’t name it .sha512, I don’t think it will end up in dotnetclichecksums feed. As of now, I believe the checksum lookup logic only reads from that feed; not sure you want me to change it to also look elsewhere. Any input on these considerations would be appreciated.

Reply:
It's not about the checksum. The hash that is included is the SHA1 hash used in the git commit. Although looking at it, it seems like it's mostly embedded into ZIPs and packages, and here we are decoupling the package, the version file, and the checksum. I was mostly asking from a "For the sake of consistency, the SDK, the runtime, and aspnetcore follow this format. Do we have a good reason to diverge from it."

Reply:
The biggest difference is that the "commit hash + version" file is found inside the zip file that is the installer for the SDK e.g. https://aka.ms/dotnet/net5/preview5/Sdk/dotnet-sdk-win-x64.zip. This URL is the vanity URL that points to the latest .NET Core 5.0 Preview 5 SDK. Because it is an installer, the build uploads it to blob storage and automatically creates the latest vanity URL for it.

The file we have is NuGet package, which is not uploaded to blob storage (it is only pushed to NuGet feeds), so no latest vanity URL is created for it. In order to no diverge form the pattern used by the SDK, we'd have to upload the NuGet package to blob storage and create the vanity URL to it. We'd also might have to include the "commit hash + version" file in the NuGet to simplify extracting the version from it (not sure where to include it because NuGet packages are typically content packages, so everything gets extracted in some fashion; if we didn't include this file, I'd have to change the update infrastructure to figure out how to extract NuGet packages (just unzip) and probably use grep+regex to get the package version from the nuspec file). With those two changes, we could then pattern based on what the SDK does. At that point, we could use the NuGet package from the blob feed instead of getting it from the NuGet feed. However, I'm not even sure if it's possible to publish a NuGet file to a blob feed instead of a NuGet feed using the Arcade SDK.

Setting aside the ability of publishing the NuGet to blob storage, the difference between what the SDK does (publish zip file with version file in it to blob storage) and what I have in this PR (publish a version file to blob storage) is only different based on which type of file is published. We will have to publish an extra file to blob storage in either case. So I think it boils down to if we can publish a NuGet to blob storage or not.

Assuming that it is possible to publish a NuGet to blob storage, I'll leave it up to you as to what you want published additionally (a NuGet or just a plain text file).

Reply:
I believe if I attach <Category>Other</Category> as metadata to the ItemsToPublishToBlobFeed item that is the NuGet package, it will be uploaded to the installers feed, which is https://dotnetcli.blob.core.windows.net/dotnet/index.json. This should actually make it easier to consume the NuGet package and update the referenced version from within dotnet-docker, assuming I can grep the version out of the nuspec easily.

Reply:
Just throwing this out there but one option might be to wrap the NuGet package in a zip file that also contains the version file.

Reply:
> Just throwing this out there but one option might be to wrap the NuGet package in a zip file that also contains the version file.

I think that's pretty much equivalent to just uploading the nupkg since the nupkg has version information in it already. Additionally, I believe that would require adding an additional build step to build an additional project that creates the zip to the dotnet/diagnostics repo. At this point, there aren't any proper installers (that I know of) that are built out of the repo, so I don't know if I want to go down that route just yet.